test/python/synthesis/test_multi_controlled_synthesis.py

```diff
@@ -0,0 +1,384 @@
+        # The bound from the documentation of synth_mcx_n_clean_m15
+        self.assertLessEqual(cx_count, 6 * num_ctrl_qubits - 6)
+
+    @data(5, 10, 15)
+    def test_mcx_1_clean_b95_cx_count(self, num_ctrl_qubits: int):
+        """Test synth_mcx_1_clean_b95 bound on CX count."""
+        synthesized_circuit = synth_mcx_1_clean_b95(num_ctrl_qubits)
+        pm = generate_preset_pass_manager(
+            optimization_level=0, basis_gates=["u", "cx"], seed_transpiler=12345
+        )
+        transpiled_circuit = pm.run(synthesized_circuit)
+        cx_count = transpiled_circuit.count_ops()["cx"]
+        # The bound from the documentation of synth_mcx_1_clean_b95
+        self.assertLessEqual(cx_count, 16 * num_ctrl_qubits - 8)
+
+    def test_c3x_cx_count(self):
+        """Test synth_c3x bound on CX count."""
+        synthesized_circuit = synth_c3x()
+        pm = generate_preset_pass_manager(
+            optimization_level=0, basis_gates=["u", "cx"], seed_transpiler=12345
+        )
+        transpiled_circuit = pm.run(synthesized_circuit)
+        cx_count = transpiled_circuit.count_ops()["cx"]
+        # The bound from the default construction for C3X
+        self.assertLessEqual(cx_count, 14)
+
+    def test_c4x_cx_count(self):
+        """Test synth_c4x bound on CX count."""
+        synthesized_circuit = synth_c4x()
+        pm = generate_preset_pass_manager(
+            optimization_level=0, basis_gates=["u", "cx"], seed_transpiler=12345
+        )
+        transpiled_circuit = pm.run(synthesized_circuit)
+        cx_count = transpiled_circuit.count_ops()["cx"]
+        # The bound from the default constuction for C4X
+        self.assertLessEqual(cx_count, 36)
+
+    @combine(
+        num_ctrl_qubits=[4, 5, 8, 10, 13, 15],
+        base_gate=[RXGate(0.123), RYGate(0.456), RZGate(0.789)],
+    )
+    def test_mc_rotation_gates_cx_count(self, num_ctrl_qubits: int, base_gate: Gate):
+        """Test bounds on the number of CX gates for mcrx / mcry / mcrz."""
+        qc = QuantumCircuit(num_ctrl_qubits + 1)
+        qc.append(base_gate.control(num_ctrl_qubits), range(num_ctrl_qubits + 1))
+        pm = generate_preset_pass_manager(
+            optimization_level=0, basis_gates=["u", "cx"], seed_transpiler=12345
+        )
+        transpiled_circuit = pm.run(qc)
+        cx_count = transpiled_circuit.count_ops()["cx"]
+        # The synthesis of mcrx/mcry/mcrz gates uses _mcsu2_real_diagonal.
+        # The bounds are given in arXiv:2302.06377, Theorem 3.
+        # In practice, we actually get better bounds for small values of num_ctrl_qubits.
+        expected_cx_count = 16 * (num_ctrl_qubits + 1) - 40
+        self.assertLessEqual(cx_count, expected_cx_count)
+
+    @data(5, 8, 10, 13, 15)
+    def test_mcx_noaux_v24_cx_count(self, num_ctrl_qubits: int):
+        """Test synth_mcx_noaux_v24 bound on CX count."""
+        synthesized_circuit = synth_mcx_noaux_v24(num_ctrl_qubits)
+        pm = generate_preset_pass_manager(
```

Comment:
You could consider moving the definition of this pass manager into a `def setUp(self)`, since it's always the same definition 🙂 I.e. something like
```python
def setUp(self):
    super().setUp()
    self.pm = generate_preset_pass_manager(...)
```
and then just use `self.pm.run` in the tests

Reply:
This is much cleaner indeed.